lms/lmsweb/views.py

```diff
@@ -208,6 +208,50 @@ def share():
     return fail(400, f'Unknown or unset act value "{act}".')
 
 
+@webapp.route('/notes/<int:user_id>', methods=['GET', 'POST'])
+@login_required
+def note(user_id: int):
+    if not current_user.role.is_manager:
+        return fail(403, "You aren't allowed to access this page.")
+
+    act = request.args.get('act') or request.json.get('act')
+
+    user = User.get_or_none(User.id == user_id)
+    if user is None:
+        return fail(404, f'No such user {user_id}.')
+
+    if act == 'delete':
+        note_id = int(request.args.get('noteId'))
+        note_ = Note.get_or_none(Note.id == note_id)
+        if (
+            note_.creator.id != current_user.id
+            and note_.is_private
```

Comment:
Why is this condition needed?

Reply:
In order to check if a manager or a staff tries to delete a private note of another manager or staff